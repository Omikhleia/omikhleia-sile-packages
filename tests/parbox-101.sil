\begin[papersize=a6]{document}
\neverindent
\nofolios
\script[src=packages/parbox]
\script[src=packages/rules]
\script[src=packages/lorem]
% An end-to-end test covering all alignement cases (top, bottom, middle)
% with all alignement options (no strut, character, rule strut)
% and with varying paddings on the last test.

\define[command=checkguide]{\rebox[width=0]{\hrule[width=100%lw, height=0.1pt]}}
\checkguide{}(1) \parbox[width=25%fw, valign=top, border=0.5pt]{para one\par
\lorem[words=5]\par
para two} \hfill \parbox[width=25%fw, valign=top, strut=character, border=0.5pt]{para one\par
\lorem[words=5]\par
para two} \hfill \parbox[width=25%fw, valign=top, strut=rule, border=0.5pt]{para one\par
\lorem[words=5]\par
para two}

\smallskip

\checkguide{}(2) \parbox[width=25%fw, valign=bottom, border=0.5pt]{para one\par
\lorem[words=5]\par
para two} \hfill \parbox[width=25%fw, valign=bottom, strut=character, border=0.5pt]{para one\par
\lorem[words=5]\par
para two} \hfill \parbox[width=25%fw, valign=bottom, strut=rule, border=0.5pt]{para one\par
\lorem[words=5]\par
para two}

\smallskip

\checkguide{}(3) \parbox[width=25%fw, valign=middle, border=0.5pt]{para one\par
\lorem[words=5]\par
para two} \hfill \parbox[width=25%fw, valign=middle, strut=character, border=0.5pt]{para one\par
\lorem[words=5]\par
para two} \hfill \parbox[width=25%fw, valign=middle, strut=rule, border=0.5pt]{para one\par
\lorem[words=5]\par
para two}

\bigskip

\checkguide{}(4) \parbox[width=25%fw, valign=top, border=0.5pt, padding=2pt 7pt 0pt 0pt]{para one\par
para two} \hfill \parbox[width=25%fw, valign=top, strut=character, border=0.5pt, padding=3pt 2pt 0pt 0pt]{para one\par
para two} \hfill \parbox[width=25%fw, valign=top, strut=rule, border=0.5pt, padding=14pt 3pt 0pt 0pt]{para one\par
para two}

\checkguide{}(5) \parbox[width=25%fw, valign=bottom, border=0.5pt, padding=2pt 7pt 0pt 0pt]{para one\par
para two} \hfill \parbox[width=25%fw, valign=bottom, strut=character, border=0.5pt, padding=3pt 2pt 0pt 0pt]{para one\par
para two} \hfill \parbox[width=25%fw, valign=bottom, strut=rule, border=0.5pt, padding=14pt 3pt 0pt 0pt]{para one\par
para two}

\checkguide{}(6) \parbox[width=25%fw, valign=middle, border=0.5pt, padding=2pt 7pt 0pt 0pt]{para one\par
para two} \hfill \parbox[width=25%fw, valign=middle, strut=character, border=0.5pt, padding=3pt 2pt 0pt 0pt]{para one\par
para two} \hfill \parbox[width=25%fw, valign=middle, strut=rule, border=0.5pt, padding=14pt 3pt 0pt 0pt]{para one\par
para two}

\end{document}