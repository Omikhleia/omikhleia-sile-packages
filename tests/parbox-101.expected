Set paper size 	297.6377985	419.5275636
Begin page
Mx 	14.8819
My 	32.9764
Draw line	14.8819	32.8764	267.8740	0.1000
Set font 	Gentium Plus;10;400;;normal;;LTR
T	11 w=3.1689	(()
Mx 	18.0508
T	20 w=4.6924	(1)
Mx 	22.7432
T	12 w=3.1689	())
Draw line	28.5547	28.2889	66.9685	0.5000
Draw line	28.5547	71.3201	66.9685	0.5000
Draw line	28.5547	28.2889	0.5000	43.5313
Draw line	95.5232	28.2889	0.5000	43.5313
Mx 	28.5547
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	82 81 72 w=15.1660	(one)
Mx 	28.5547
My 	44.9764
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	54.8149
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	81.9246
T	71 82 w=10.2295	(do)
Mx 	92.1541
T	16 w=3.3691	(-)
Mx 	28.5547
My 	56.9764
T	79 82 85 w=11.6992	(lor)
Mx 	42.8987
T	86 76 87 w=10.0146	(sit)
Mx 	55.5580
T	68 80 72 87 w=20.6836	(amet)
Mx 	28.5547
My 	68.9764
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	87 90 82 w=15.5322	(two)
Draw line	122.1711	24.8221	66.9685	0.5000
Draw line	122.1711	71.4178	66.9685	0.5000
Draw line	122.1711	24.8221	0.5000	47.0957
Draw line	189.1396	24.8221	0.5000	47.0957
Mx 	122.1711
My 	32.9764
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	82 81 72 w=15.1660	(one)
Mx 	122.1711
My 	44.9764
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	148.4312
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	175.5409
T	71 82 w=10.2295	(do)
Mx 	185.7704
T	16 w=3.3691	(-)
Mx 	122.1711
My 	56.9764
T	79 82 85 w=11.6992	(lor)
Mx 	136.5150
T	86 76 87 w=10.0146	(sit)
Mx 	149.1743
T	68 80 72 87 w=20.6836	(amet)
Mx 	122.1711
My 	68.9764
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	87 90 82 w=15.5322	(two)
Draw line	215.7874	20.9764	66.9685	0.5000
Draw line	215.7874	72.5764	66.9685	0.5000
Draw line	215.7874	20.9764	0.5000	52.1000
Draw line	282.7559	20.9764	0.5000	52.1000
Mx 	215.7874
My 	32.9764
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8709
T	82 81 72 w=15.1660	(one)
Mx 	215.7874
My 	44.9764
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	242.0475
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	269.1573
T	71 82 w=10.2295	(do)
Mx 	279.3868
T	16 w=3.3691	(-)
Mx 	215.7874
My 	56.9764
T	79 82 85 w=11.6992	(lor)
Mx 	230.1313
T	86 76 87 w=10.0146	(sit)
Mx 	242.7907
T	68 80 72 87 w=20.6836	(amet)
Mx 	215.7874
My 	68.9764
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8708
T	87 90 82 w=15.5322	(two)
Mx 	14.8819
My 	124.5764
Draw line	14.8819	124.4764	267.8740	0.1000
T	11 w=3.1689	(()
Mx 	18.0508
T	21 w=4.6924	(2)
Mx 	22.7432
T	12 w=3.1689	())
Draw line	28.5547	83.8889	66.9685	0.5000
Draw line	28.5547	126.9201	66.9685	0.5000
Draw line	28.5547	83.8889	0.5000	43.5313
Draw line	95.5232	83.8889	0.5000	43.5313
Mx 	28.5547
My 	88.5764
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	82 81 72 w=15.1660	(one)
Mx 	28.5547
My 	100.5764
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	54.8149
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	81.9246
T	71 82 w=10.2295	(do)
Mx 	92.1541
T	16 w=3.3691	(-)
Mx 	28.5547
My 	112.5764
T	79 82 85 w=11.6992	(lor)
Mx 	42.8987
T	86 76 87 w=10.0146	(sit)
Mx 	55.5580
T	68 80 72 87 w=20.6836	(amet)
Mx 	28.5547
My 	124.5764
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	87 90 82 w=15.5322	(two)
Draw line	122.1711	80.4221	66.9685	0.5000
Draw line	122.1711	127.0178	66.9685	0.5000
Draw line	122.1711	80.4221	0.5000	47.0957
Draw line	189.1396	80.4221	0.5000	47.0957
Mx 	122.1711
My 	88.5764
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	82 81 72 w=15.1660	(one)
Mx 	122.1711
My 	100.5764
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	148.4312
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	175.5409
T	71 82 w=10.2295	(do)
Mx 	185.7704
T	16 w=3.3691	(-)
Mx 	122.1711
My 	112.5764
T	79 82 85 w=11.6992	(lor)
Mx 	136.5150
T	86 76 87 w=10.0146	(sit)
Mx 	149.1743
T	68 80 72 87 w=20.6836	(amet)
Mx 	122.1711
My 	124.5764
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	87 90 82 w=15.5322	(two)
Draw line	215.7874	76.5764	66.9685	0.5000
Draw line	215.7874	128.1764	66.9685	0.5000
Draw line	215.7874	76.5764	0.5000	52.1000
Draw line	282.7559	76.5764	0.5000	52.1000
Mx 	215.7874
My 	88.5764
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8709
T	82 81 72 w=15.1660	(one)
Mx 	215.7874
My 	100.5764
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	242.0475
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	269.1573
T	71 82 w=10.2295	(do)
Mx 	279.3868
T	16 w=3.3691	(-)
Mx 	215.7874
My 	112.5764
T	79 82 85 w=11.6992	(lor)
Mx 	230.1313
T	86 76 87 w=10.0146	(sit)
Mx 	242.7907
T	68 80 72 87 w=20.6836	(amet)
Mx 	215.7874
My 	124.5764
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8708
T	87 90 82 w=15.5322	(two)
Mx 	14.8819
My 	157.9764
Draw line	14.8819	157.8764	267.8740	0.1000
T	11 w=3.1689	(()
Mx 	18.0508
T	22 w=4.6924	(3)
Mx 	22.7432
T	12 w=3.1689	())
Draw line	28.5547	136.4608	66.9685	0.5000
Draw line	28.5547	179.4920	66.9685	0.5000
Draw line	28.5547	136.4608	0.5000	43.5313
Draw line	95.5232	136.4608	0.5000	43.5313
Mx 	28.5547
My 	141.1483
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	82 81 72 w=15.1660	(one)
Mx 	28.5547
My 	153.1483
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	54.8149
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	81.9246
T	71 82 w=10.2295	(do)
Mx 	92.1541
T	16 w=3.3691	(-)
Mx 	28.5547
My 	165.1483
T	79 82 85 w=11.6992	(lor)
Mx 	42.8987
T	86 76 87 w=10.0146	(sit)
Mx 	55.5580
T	68 80 72 87 w=20.6836	(amet)
Mx 	28.5547
My 	177.1483
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	87 90 82 w=15.5322	(two)
Draw line	122.1711	134.6785	66.9685	0.5000
Draw line	122.1711	181.2742	66.9685	0.5000
Draw line	122.1711	134.6785	0.5000	47.0957
Draw line	189.1396	134.6785	0.5000	47.0957
Mx 	122.1711
My 	142.8328
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	82 81 72 w=15.1660	(one)
Mx 	122.1711
My 	154.8328
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	148.4312
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	175.5409
T	71 82 w=10.2295	(do)
Mx 	185.7704
T	16 w=3.3691	(-)
Mx 	122.1711
My 	166.8328
T	79 82 85 w=11.6992	(lor)
Mx 	136.5150
T	86 76 87 w=10.0146	(sit)
Mx 	149.1743
T	68 80 72 87 w=20.6836	(amet)
Mx 	122.1711
My 	178.8328
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	87 90 82 w=15.5322	(two)
Draw line	215.7874	132.1764	66.9685	0.5000
Draw line	215.7874	183.7764	66.9685	0.5000
Draw line	215.7874	132.1764	0.5000	52.1000
Draw line	282.7559	132.1764	0.5000	52.1000
Mx 	215.7874
My 	144.1764
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8709
T	82 81 72 w=15.1660	(one)
Mx 	215.7874
My 	156.1764
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	242.0475
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	269.1573
T	71 82 w=10.2295	(do)
Mx 	279.3868
T	16 w=3.3691	(-)
Mx 	215.7874
My 	168.1764
T	79 82 85 w=11.6992	(lor)
Mx 	230.1313
T	86 76 87 w=10.0146	(sit)
Mx 	242.7907
T	68 80 72 87 w=20.6836	(amet)
Mx 	215.7874
My 	180.1764
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8708
T	87 90 82 w=15.5322	(two)
Mx 	14.8819
My 	222.7764
Draw line	14.8819	222.6764	267.8740	0.1000
T	11 w=3.1689	(()
Mx 	18.0508
T	23 w=4.6924	(4)
Mx 	22.7432
T	12 w=3.1689	())
Draw line	28.5547	216.0889	66.9685	0.5000
Draw line	28.5547	244.1201	66.9685	0.5000
Draw line	28.5547	216.0889	0.5000	28.5313
Draw line	95.5232	216.0889	0.5000	28.5313
Mx 	28.5547
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	82 81 72 w=15.1660	(one)
Mx 	28.5547
My 	234.7764
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	87 90 82 w=15.5322	(two)
Draw line	122.1711	211.6221	66.9685	0.5000
Draw line	122.1711	239.2178	66.9685	0.5000
Draw line	122.1711	211.6221	0.5000	28.0957
Draw line	189.1396	211.6221	0.5000	28.0957
Mx 	122.1711
My 	222.7764
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	82 81 72 w=15.1660	(one)
Mx 	122.1711
My 	234.7764
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	87 90 82 w=15.5322	(two)
Draw line	215.7874	196.7764	66.9685	0.5000
Draw line	215.7874	241.3764	66.9685	0.5000
Draw line	215.7874	196.7764	0.5000	45.1000
Draw line	282.7559	196.7764	0.5000	45.1000
Mx 	215.7874
My 	222.7764
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8709
T	82 81 72 w=15.1660	(one)
Mx 	215.7874
My 	234.7764
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8708
T	87 90 82 w=15.5322	(two)
Mx 	14.8819
My 	283.1201
Draw line	14.8819	283.0201	267.8740	0.1000
T	11 w=3.1689	(()
Mx 	18.0508
T	24 w=4.6924	(5)
Mx 	22.7432
T	12 w=3.1689	())
Draw line	28.5547	264.4326	66.9685	0.5000
Draw line	28.5547	292.4639	66.9685	0.5000
Draw line	28.5547	264.4326	0.5000	28.5313
Draw line	95.5232	264.4326	0.5000	28.5313
Mx 	28.5547
My 	271.1201
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	82 81 72 w=15.1660	(one)
Mx 	28.5547
My 	283.1201
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	87 90 82 w=15.5322	(two)
Draw line	122.1711	259.9658	66.9685	0.5000
Draw line	122.1711	287.5615	66.9685	0.5000
Draw line	122.1711	259.9658	0.5000	28.0957
Draw line	189.1396	259.9658	0.5000	28.0957
Mx 	122.1711
My 	271.1201
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	82 81 72 w=15.1660	(one)
Mx 	122.1711
My 	283.1201
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	87 90 82 w=15.5322	(two)
Draw line	215.7874	245.1201	66.9685	0.5000
Draw line	215.7874	289.7201	66.9685	0.5000
Draw line	215.7874	245.1201	0.5000	45.1000
Draw line	282.7559	245.1201	0.5000	45.1000
Mx 	215.7874
My 	271.1201
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8709
T	82 81 72 w=15.1660	(one)
Mx 	215.7874
My 	283.1201
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8708
T	87 90 82 w=15.5322	(two)
Mx 	14.8819
My 	315.7639
Draw line	14.8819	315.6639	267.8740	0.1000
T	11 w=3.1689	(()
Mx 	18.0508
T	25 w=4.6924	(6)
Mx 	22.7432
T	12 w=3.1689	())
Draw line	28.5547	301.7483	66.9685	0.5000
Draw line	28.5547	329.7795	66.9685	0.5000
Draw line	28.5547	301.7483	0.5000	28.5313
Draw line	95.5232	301.7483	0.5000	28.5313
Mx 	28.5547
My 	308.4358
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	82 81 72 w=15.1660	(one)
Mx 	28.5547
My 	320.4358
T	83 68 85 68 w=18.4375	(para)
Mx 	49.6382
T	87 90 82 w=15.5322	(two)
Draw line	122.1711	301.9660	66.9685	0.5000
Draw line	122.1711	329.5617	66.9685	0.5000
Draw line	122.1711	301.9660	0.5000	28.0957
Draw line	189.1396	301.9660	0.5000	28.0957
Mx 	122.1711
My 	313.1203
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	82 81 72 w=15.1660	(one)
Mx 	122.1711
My 	325.1203
T	83 68 85 68 w=18.4375	(para)
Mx 	143.2545
T	87 90 82 w=15.5322	(two)
Draw line	215.7874	293.4639	66.9685	0.5000
Draw line	215.7874	338.0639	66.9685	0.5000
Draw line	215.7874	293.4639	0.5000	45.1000
Draw line	282.7559	293.4639	0.5000	45.1000
Mx 	215.7874
My 	319.4639
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8709
T	82 81 72 w=15.1660	(one)
Mx 	215.7874
My 	331.4639
T	83 68 85 68 w=18.4375	(para)
Mx 	236.8708
T	87 90 82 w=15.5322	(two)
End page
Finish
